packages/compiler/src/shadow_css.ts

```diff
@@ -157,6 +157,64 @@ export class ShadowCss {
     return this._insertPolyfillRulesInCssText(cssText);
   }
 
+  /*
+   * Process styles to add scope to keyframes, this implies modifying both
+   * the names of the keyframes defined in the css file and also the css
+   * animation rules using them (animation rules using keyframes defined
+   * elsewhere need not to be modified to allow for globally defined keyframes)
+   *
+   * For example, we convert this css:
+   *
+   * .box {
+   *    animation: box-animation 1s forwards;
+   *  }
+   *
+   * @keyframes box-animation {
+   *  to {
+   *    background-color: green;
+   *  }
+   * }
+   *
+   * to this:
+   *
+   * .box {
+   *    animation: scopeName_box-animation 1s forwards;
+   *  }
+   *
+   * @keyframes scopeName_box-animation {
+   *  to {
+   *    background-color: green;
+   *  }
+   * }
+   *
+   **/
+  private _scopeLocalKeyframes(cssText: string, scopeSelector: string): string {
+    // set used to collect all keyframes names (before scoping) so that they can
+    // later used to scope the animation rules
+    const unscopedKeyframes = new Set<string>();
+
+    const scopedKeyframesCssText = processRules(cssText, rule => {
+      return {
+        ...rule,
+        selector: rule.selector.replace(
+            /^@(-webkit-)?keyframes\s([^\s]+)\s?/,
+            (_, webkit, keyframeName) => {
+              unscopedKeyframes.add(keyframeName);
+              return `@${webkit || ''}keyframes ${scopeSelector}_${keyframeName} `;
+            })
+      };
+    });
+    return processRules(
+        scopedKeyframesCssText,
+        rule => ({
+          ...rule,
+          content: rule.content.replace(
+              /(animation|animation-name)\s*:\s*([^\r\n\t\f\v;, ]+)/,
```

Comment:
Any reason not to just use `\s` to match the whitespace?

```suggestion
              /(animation|animation-name)\s*:\s*([^\s;,]+)/,
```

Reply:
sorry I am not the best with regexes and this is what regex101 suggested:
![Screenshot at 2021-06-24 22-12-05](https://user-images.githubusercontent.com/61631103/123333148-49074f00-d539-11eb-9075-5baca0965259.png)

yeah it makes total sense to just have `\s` sorry, I'll fix this :slightly_smiling_face: